flash/audio/speech_recognition/data.py

```diff
@@ -0,0 +1,250 @@
+from flash.core.utilities.imports import _TEXT_AVAILABLE
+
+if _TEXT_AVAILABLE:
+    from datasets import load_dataset
+    from transformers import Wav2Vec2CTCTokenizer, Wav2Vec2Processor
+
+
+class AudioDeserializer(Deserializer):
+
+    def __init__(self, backbone: str, max_length: int):
+        super().__init__()
+        self.backbone = backbone
+        self.tokenizer = Wav2Vec2CTCTokenizer.from_pretrained(backbone)
+        self.max_length = max_length
+
+    def deserialize(self, sample: Any) -> Tensor:
+        return self.tokenizer(sample["speech"], sampling_rate=sample["sampling_rate"][0]).input_values
+
+    @property
+    def example_input(self) -> str:
+        return "An example input"
+
+    def __getstate__(self):  # TODO: Find out why this is being pickled
+        state = self.__dict__.copy()
+        state.pop("processor")
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self.tokenizer = Wav2Vec2CTCTokenizer.from_pretrained(self.backbone)
+
+
+class AudioDataSource(DataSource):
+
+    def __init__(self, backbone: str, max_length: int = 128):
+        super().__init__()
+
+        self.backbone = backbone
+        self.processor = Wav2Vec2Processor.from_pretrained(backbone)
+        self.max_length = max_length
+
+    def _prepare_dataset(self, batch):
+        # check that all files have the correct sampling rate
+        assert (
+            len(set(batch["sampling_rate"])) == 1
+        ), f"Make sure all inputs have the same sampling rate of {self.processor.feature_extractor.sampling_rate}."
+
+        batch["input_values"] = self.processor(batch["speech"], sampling_rate=batch["sampling_rate"][0]).input_values
+
+        if not self.predicting:
+            with self.processor.as_target_processor():
+                batch["labels"] = self.processor(batch["target_text"]).input_ids
+        return batch
+
+    @staticmethod
+    def speech_file_to_array_fn(batch):
+        speech_array, sampling_rate = sf.read(batch["file"])
+        batch["speech"] = speech_array
+        batch["sampling_rate"] = sampling_rate
+        batch["target_text"] = batch["text"]
+        return batch
+
+    def load_data(
+        self,
+        data: Tuple[str, Union[str, List[str]], Union[str, List[str]]],
+        dataset: Optional[Any] = None,
+        columns: Union[List[str], Tuple[str]] = ("input_values", "labels"),
+    ) -> Union[Sequence[Mapping[str, Any]]]:
+        # file, input, target = data
+        #
+        # data_files = {}
+        stage = self.running_stage.value
+        # data_files[stage] = str(file)
+
+        dataset_dict = load_dataset("timit_asr")  # todo
+        dataset_dict = dataset_dict.map(AudioDataSource.speech_file_to_array_fn, num_proc=4)
+        dataset_dict = dataset_dict.map(partial(self._prepare_dataset), batched=True)
+        dataset_dict.set_format("torch", columns=columns)
+
+        return dataset_dict[stage]
```

Comment:
Not sure this will work for predict. Might want to implement a `predict_load_data`.